profiles/views.py

```diff
@@ -0,0 +1,38 @@
+# profiles/views.py
+from django.shortcuts import render, redirect, get_object_or_404
+from django.contrib.auth.decorators import login_required
+from .forms import ProfileForm
+from django.http import Http404
+from .models import UserProfile
+from events.models import Event
+
+
+@login_required
+def view_profile(request, userprofile_id):
+    user_profile = get_object_or_404(UserProfile, pk=userprofile_id)
+    events = user_profile.user.event_set.filter(is_active=True)
+    context = {"user_profile": user_profile, "events": events}
+
+    return render(request, "profiles/view_profile.html", context)
+
+
+@login_required
+def edit_profile(request):
+    user_profile = request.user.userprofile
+
+    if request.method == "POST":
+        form = ProfileForm(request.POST, request.FILES, instance=user_profile)
```

Comment:
Here we need the backend validation to allow only the user him/herself to edit the profile. Otherwise anyone can edit the profile through url. A test case is also needed for this.

Reply:
Oh I see your logic here. Sorry never mind.